src/echo-common/src/poe-zone-tracker-service.ts

```diff
@@ -0,0 +1,465 @@
+          )
+        }
+
+        this.currentZoneDelta.poeZoneInstance.areaLevel = event.areaLevel
+        this.currentZoneDelta.poeZoneInstance.zoneSeed = event.seed
+        this.currentZoneDelta.poeZoneInstance.areaTag = event.areaTag
+
+        //Determine if currentZoneDelta is a revisit of an old zone
+        //Has server, arealevel, seed, areatag and
+        let zoneHash: string =
+          this.currentZoneDelta.poeZoneInstance.zoneServer +
+          '|' +
+          this.currentZoneDelta.poeZoneInstance.areaTag +
+          '|' +
+          this.currentZoneDelta.poeZoneInstance.areaLevel.toString() +
+          '|' +
+          this.currentZoneDelta.poeZoneInstance.zoneSeed.toString()
+
+        if (
+          event.areaTag.toLowerCase().includes('hideout') ||
+          event.areaTag.toLowerCase().includes('town')
+        ) {
+          this.currentZoneDelta.poeZoneInstance.isZoneHideoutOrTown = true
+          zoneHash = zoneHash + '|' + event.time.getTime().toString()
+        } else {
+          this.currentZoneDelta.poeZoneInstance.isZoneHideoutOrTown = false
+        }
+
+        this.currentZoneDelta.poeZoneInstance.zoneHash = zoneHash
+
+        const existingZone = this.zones.get(zoneHash)
+        if (existingZone) {
+          this.currentZoneDelta.poeZoneInstance = existingZone
+          //TODO IF EXISTING ZONE IS FOUND, DO SOME LOGIC BASED ON DATE TO DETERMINE IF IT IS CLOSE ENOUGH
+          existingZone.zoneDeltas.push(this.currentZoneDelta)
+          this.zones.set(zoneHash, existingZone)
+        } else {
+          this.currentZoneDelta.poeZoneInstance.zoneDeltas.push(this.currentZoneDelta)
+          this.zones.set(zoneHash, this.currentZoneDelta.poeZoneInstance)
+        }
+      })
+  }
+
+  private zoneEntranceEventSubscribe() {
+    this.logEvents$
+      .pipe(
+        filter((e) => e.type == 'ZoneEntranceEvent'),
+        map((e) => e as PoeZoneEntranceEvent)
+      )
+      .subscribe((event) => {
+        if (this.currentZoneDelta === undefined) {
+          //Emit as event
+          throw new Error(
+            'Something went wrong. currentZoneDelta not instansiated correctly in ZoneEntranceEvent'
+          )
+        }
+
+        if (!this.currentZoneDelta.startLoadTime) {
+          //Emit as event
+          throw new Error(
+            'Something went wrong. currentZoneDelta.startLoadTime not set correctly in ZoneEntranceEvent'
+          )
+        }
+
+        this.currentZoneDelta.poeZoneInstance.zoneName = event.location
+        this.currentZoneDelta.enterTime = event.time
+        this.currentZoneDelta.enterSystemTime = event.systemUptime
+
+        const loadStart = this.currentZoneDelta.startLoadTime
+        this.currentZoneDelta.doneLoadTime = event.systemUptime
+        this.currentZoneDelta.deltaLoadTime = event.systemUptime - loadStart
+
+        if (this.characterService != undefined) {
+          const charName = 'BlaesendeBruno' //TODO from settings
```

Comment:
We probably want to use the character with `current` set to true from the character list as a general solution, or make this class more of a util than a `context()` object. If it was constructed by the plugins themselves they could pass in and manage what character to track themselves.

Reply:
I removed the character based stuff for now, as it was not properly ready to be merged into main and be used elsewhere